From a6803dc717c40978607484e0ab4cda4a7aa2e5f7 Mon Sep 17 00:00:00 2001
From: Chris Thrasher <chrisjthrasher@gmail.com>
Date: Thu, 15 Feb 2024 11:40:21 -0700
Subject: [PATCH] Disable compiler warnings by default

Unconditionally enabling -Werror is a heavy-handed approach and not
ideal when shipping code to be used by many third parties. In fact
it's even better to disable warnings by default since compiler
warnings are not hard requirements. They're merely a development
tool that developers should opt into.
---
 CMakeLists.txt    | 3 ++-
 CMakePresets.json | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d47b08..b800e66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,8 @@ find_package(rclcpp REQUIRED)
 find_package(tcb_span REQUIRED)
 find_package(tl_expected REQUIRED)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
+option(RSL_ENABLE_WARNINGS "Enable compiler warnings" OFF)
+if(RSL_ENABLE_WARNINGS AND CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
     add_compile_options(-Werror -Wall -Wextra -Wpedantic -Wshadow -Wconversion -Wsign-conversion -Wold-style-cast)
 endif()
 
diff --git a/CMakePresets.json b/CMakePresets.json
index c7565ae..2d35c1f 100644
--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -18,7 +18,8 @@
                 "CMAKE_EXE_LINKER_FLAGS": "-fuse-ld=lld",
                 "CMAKE_MODULE_LINKER_FLAGS": "-fuse-ld=lld",
                 "CMAKE_SHARED_LINKER_FLAGS": "-fuse-ld=lld",
-                "RSL_BUILD_TESTING": "ON"
+                "RSL_BUILD_TESTING": "ON",
+                "RSL_ENABLE_WARNINGS": "ON"
             },
             "warnings": {
                 "unusedCli": false
